clang-tools/ClangKast/ClangKast.h

```diff
@@ -0,0 +1,139 @@
+#ifndef CLANGKAST_H_
+#define CLANGKAST_H_
+
+enum class Sort {
+  ACCESSSPECIFIER,
+  AEXPR,
+  ASTMT,
+  ATYPE,
+  ATYPERESULT,
+  AUTOSPECIFIER,
+  BASESPECIFIER,
+  BOOL,
+  BRACEINIT,
+  CABSLOC,
+  CAPTURE,
+  CAPTUREDEFAULT,
+  CAPTUREKIND,
+  CATCHDECL,
+  CHARKIND,
+  CID,
+  CLASSKEY,
+  CTORINIT,
+  DECL,
+  DECLARATOR,
+  DESTRUCTORID,
+  ENUMERATOR,
+  EXCEPTIONSPEC,
+  EXPR,
+  EXPRESSIONLIST,
+  EXPRLOC,
+  FLOAT,
+  FUNCTIONSPECIFIER,
+  INIT,
+  INT,
+  K,
+  KITEM,
+  LIST,
+  NAME,
+  NAMESPACE,
+  NNS,
+  NNSSPECIFIER,
+  NONAME,
+  OPID,
+  QUALIFIER,
+  REFQUALIFIER,
+  RESOLVEDEXPR,
+  SPECIFIER,
+  STMT,
+  STORAGECLASSSPECIFIER,
+  STRICTLIST,
+  STRICTLISTRESULT,
+  STRING,
+  TAG,
+  TEMPLATEARGUMENT,
+  TEMPLATEPARAMETER,
+  THIS,
+  TYPEID,
+  TYPESPECIFIER,
+  UNNAMEDCID,
+};
+
+std::ostream& operator<<(std::ostream & os, const Sort & sort);
+
+class Kast {
+public:
+
+  // Add a Kast::Node to the KAST. The next nodes added should be the children (bredth-first).
+  template <class T>
+  static void add(const T & node);
+
+  // Print the KAST.
+  static void print();
+
+  class Node {
```

Comment:
This class should have a virtual destructor (see https://stackoverflow.com/a/461224/6209703)